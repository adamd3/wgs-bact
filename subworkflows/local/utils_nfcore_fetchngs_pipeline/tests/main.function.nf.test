
nextflow_function {

    name "Test Functions"
    script "subworkflows/local/utils_nfcore_fetchngs_pipeline/main.nf"
    tag 'subworkflows'
    tag 'utils_nfcore_fetchngs_pipeline'
    tag 'subworkflows/utils_nfcore_fetchngs_pipeline'

// isSraId(input)
// isSynapseId(input)
// sraCheckENAMetadataFields(ena_metadata_fields)
// synapseShowToMap(synapse_file)
// synapseSampleNameFromFastQ(input_file, pattern)

    test("Function sraCurateSamplesheetWarn") {

        function "sraCurateSamplesheetWarn"

        then {
            assert function.success
            assert snapshot(function.result).match()
        }
    }

    test("Function synapseCurateSamplesheetWarn") {

        function "synapseCurateSamplesheetWarn"

        then {
            assert function.success
            assert snapshot(function.result).match()
        }
    }
}
