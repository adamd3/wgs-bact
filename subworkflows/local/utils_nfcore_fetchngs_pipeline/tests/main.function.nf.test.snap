{
    "Function sraCurateSamplesheetWarn": {
        "content": null,
        "timestamp": "2024-01-19T16:37:05.668505"
    },
    "Function sraCheckENAMetadataFields [success]": {
        "content": null,
        "timestamp": "2024-01-19T16:36:51.121603"
    },
    "Function isSraId": {
        "content": [
            true
        ],
        "timestamp": "2024-01-19T16:36:43.68958"
    }
}